azure-pipelines.yml

```diff
@@ -45,8 +45,12 @@ jobs:
         versionSpec: '3.9'
     - bash: |
         # Include pytest compatibility with mypy
-        pip install pytest flake8 mypy==0.782
+        pip install pytest flake8 mypy==0.782 black
```

Comment:
```suggestion
        pip install pytest flake8 mypy==0.782 black==21.6b0
```

Reply:
If we pin a specific version of black in the CI, we should also pin it in `doc/developers/contributing.rst`.

However I am not sure it's worth the added complexity. I am pretty sure that devs will often have the latest version of black in their dev env.

I would be in favor of trying without pinning. If it proves annoying we can always decide to pin a specific version later until the black format spec is declared final.

Reply:
After having used black on some other projects for a while, I find that that if one doesn't pin black version one will get in trouble eventually. For instance my current scikit-learn dev env is from 2020 and the version there produces a slightly different formatting to version `21.6b0` . Similarly there might a change that the next release will not be fully compatible etc.  So it's not like flake8 where the version doesn't matter.

Reply:
And for sure the pre-commit version and the CI version need to match exactly. Otherwise we will get CI failure for each new black release.

Reply:
Alright, I am fine with pinning, then (if others agree).